flink-runtime/src/main/java/org/apache/flink/runtime/net/ConnectionUtils.java

```diff
@@ -180,7 +180,41 @@ public static InetAddress findConnectingAddress(InetSocketAddress targetAddress,
 		}
 	}
 
+	/**
+	 * This utility method tries to connect to the JobManager using the InetAddress returned by
+	 * InetAddress.getLocalHost(). The purpose of the utility is to have a final try connecting to
+	 * the target address using the LocalHost before using the address returned.
+	 * We do a second try because the JM might have been unavailable during the first check.
+	 *
+	 * @param preliminaryResult The address detected by the heuristic
+	 * @return either the preliminaryResult or the address returned by InetAddress.getLocalHost() (if
+	 * 			we are able to connect to targetAddress from there)
+	 */
+	private static InetAddress tryLocalHostBeforeReturning(InetAddress preliminaryResult, SocketAddress targetAddress, boolean logging) throws IOException {
+		InetAddress localhostName = InetAddress.getLocalHost();
+		if(tryToConnect(localhostName, targetAddress, AddressDetectionState.LOCAL_HOST.getTimeout(), logging)) {
```

Comment:
I think we can use a bit higher timeout here. 200ms is probably mostly enough, but why make it so short here? If it succeeds fast, it does not add delay, and it it does not succeed we went through the other interfaces already anyways and better spend another second to make sure we get it right.